src/raylib.cc

```diff
@@ -629,8 +629,12 @@ PyObject* wait_for_next_task(PyObject* self, PyObject* args) {
   if (!PyArg_ParseTuple(args, "O&", &PyObjectToWorker, &worker)) {
     return NULL;
   }
-  Task* task = worker->receive_next_task();
-  return PyCapsule_New(static_cast<void*>(task), "task", NULL); // This task is owned by the C++ worker class, so we do not deallocate it.
+  if (std::unique_ptr<Task> task = worker->receive_next_task()) {
+    PyObject* pyobj = PyCapsule_New(task.get(), "task", TaskCapsule_Destructor);
```

Comment:
Ownership needs to be released once it has been already transferred, not before that. Otherwise an exception could cause a memory leak. I know you haven't turned on exceptions, but that doesn't preclude writing exception-safe code.